apps/projection/src/app/ui/card/card.component.html

```diff
@@ -1,29 +1,24 @@
-<div class="border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3" [class]="customClass">
-  <img
-    *ngIf="type === CardType.TEACHER"
-    src="assets/img/teacher.png"
-    width="200px"
-  />
-  <img
-    *ngIf="type === CardType.STUDENT"
-    src="assets/img/student.webp"
-    width="200px"
-  />
-
+<div
+  class="border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3"
+  [class]="customClass">
+  <ng-content select="img"></ng-content>
   <section>
-    <app-list-item
-      *ngFor="let item of list"
-      [name]="item.firstname"
-      [id]="item.id"
-      [type]="type"
-    >
+    <!-- <ng-container [ngTemplateOutlet]="childTemplate"></ng-container> -->
+    <app-list-item *ngFor="let item of list">
+      <ng-template #listTemplate>
+        <div class="border border-grey-300 py-1 px-2 flex justify-between">
+          {{ item.firstname ? item.firstname : item.name }}
```

Comment:
You are getting there but not completely. Having conditions inside a shared component should trigger a warning that the pattern use is not the right one. You will end up adding condition each time you add a new component or that the product team wants a new design. 

What is displayed on the row should be done inside the parent element (TeacherCard, CityCard, ... )? To do that, use ngTemplateOutlet with context. 

The delete button should stay inside the ListItemComponent. The way you did it makes the ListItem useless. 

NgTemplateOutlet should be inside this component instead of the ng-template and the template should be inside the parent element. And ListItem wlll be call inside the template in the parent element. Hope this make sense but your code is getting better. Keep up the good work 💪

Reply:
I have changed the above as per the review comments and took help from other PR as well as from this video. [Link](https://www.youtube.com/watch?v=_v_MEm9iV-c)